chain/ethereum/src/network.rs

```diff
@@ -48,57 +71,96 @@ impl EthereumNetworkAdapter {
 
 #[derive(Debug, Clone)]
 pub struct EthereumNetworkAdapters {
-    pub adapters: Vec<EthereumNetworkAdapter>,
+    chain_id: ChainId,
+    manager: ProviderManager<EthereumNetworkAdapter>,
     call_only_adapters: Vec<EthereumNetworkAdapter>,
     // Percentage of request that should be used to retest errored adapters.
     retest_percent: f64,
 }
 
 impl Default for EthereumNetworkAdapters {
     fn default() -> Self {
-        Self::new(None)
+        Self {
+            chain_id: "".into(),
+            manager: ProviderManager::default(),
+            call_only_adapters: vec![],
+            retest_percent: DEFAULT_ADAPTER_ERROR_RETEST_PERCENT,
+        }
     }
 }
```

Comment:
I usually add these types of defaults when the setup is not super straightforward so that anyone using it doesn't have to dive into how the struct works to use. The main issue with ChainClient is that in some cases we do need to use both, I could probably add a third option that explicitly uses firehose AND rpc to remove that bound